mm2src/coins/z_coin.rs

```diff
@@ -777,7 +791,15 @@ impl SwapOps for ZCoin {
             .into_script();
         let selfi = self.clone();
         let fut = async move {
-            let tx_fut = z_p2sh_spend(&selfi, tx, time_lock, SEQUENCE_FINAL, redeem_script, script_data);
+            let tx_fut = z_p2sh_spend(
+                &selfi,
+                tx,
+                time_lock,
+                SEQUENCE_FINAL,
+                redeem_script,
+                script_data,
+                key_pair.private().secret.as_slice(),
```

Comment:
It's in the `async move` block so it will require an additional boilerplate to make an owned version of `htlc_privkey` anyway. I have to keep it this way for now.